src/hammer-vlsi/hammer_vlsi/cli_driver.py

```diff
@@ -856,6 +859,77 @@ def auto_action(driver: HammerDriver, append_error_func: Callable[[str], None])
 
         return driver, errors
 
+    @staticmethod
+    def make_build_inputs(driver: HammerDriver, append_error_func: Callable[[str], None]) -> Optional[dict]:
+        """
+        TODO
+        """
+        build_system_key = "vlsi.core.build_system"
+        build_system = str(driver.database.get_setting(build_system_key, "none"))
+        dependency_graph = driver.get_hierarchical_dependency_graph()
+        if build_system == "none":
+            pass
+        elif build_system == "make":
+            makefile = os.path.join(driver.obj_dir, "hammer.d")
+            default_dependencies = driver.options.project_configs + driver.options.environment_configs
+            default_dependencies.extend(list(driver.database.get_setting("synthesis.inputs.input_files", [])))
+            # Resolve the canonical path for each dependency
+            default_dependencies = [os.path.realpath(x) for x in default_dependencies]
+            output = ""
+            output += "HAMMER_EXEC ?= {}\n".format(os.path.realpath(sys.argv[0]))
+            output += "HAMMER_DEPENDENCIES ?= {}\n\n".format(" ".join(default_dependencies))
+            if not dependency_graph:
+                # Flat flow
+                # TODO
+                raise NotImplementedError("Haven't implemented this yet")
+            else:
+                # Hierarchical flow
+                for node, edges in dependency_graph.items():
+                    env_confs = " ".join(["-e " + os.path.realpath(x) for x in driver.options.environment_configs])
+                    syn_in = " ".join(["-p " + os.path.realpath(x) for x in driver.options.project_configs])
+                    in_edges = edges[0]
+                    out_edges = edges[1]
+                    deps = "$(HAMMER_DEPENDENCIES)"
+                    if len(out_edges) > 0:
+                        deps = " ".join(["par-" + x for x in out_edges])
+
+                    # TODO make this DRY
+                    obj_dir = os.path.realpath(driver.obj_dir)
+                    syn_run_dir = os.path.join(obj_dir, "syn-" + node)
+                    par_run_dir = os.path.join(obj_dir, "par-" + node)
+                    syn_timestamp = os.path.join(syn_run_dir, "timestamp")
+                    par_timestamp = os.path.join(par_run_dir, "timestamp")
+
+                    syn_out = os.path.join(syn_run_dir, "syn-output.json")
```

Comment:
Would be good if we can standardize these names somewhere (right now they're just scattered over the code manually)

Reply:
Agree. I tried calling `par_tool.run_dir` but this all happens before the driver sets up the syn and par tools, so those are `None`. Not sure the best way to do it. I'm inclined to leave it as a TODO for this PR, but if you can think of a clever, low-effort way to get it in I'm all ears.

Reply:
TODO is fine for now. Maybe file a `DeveloperSupport` issue?

Reply:
OK, filed #450